fbpic/boundaries/boundary_communicator.py

```diff
@@ -583,7 +731,18 @@ def gather_grid_array(self, array, root = 0):
             gathered_array = None
         # Shortcut for the guard cells
```

Comment:
For more conciseness, I would do 
```python
n_left = self.n_guard
n_right = self.n_guard
# Remove n_damp cells from the left proc's output region
if self.left_proc is None:
    n_left += self.n_damp
# Remove n_damp cells from the right proc's output region
if self.right_proc is None:
   n_right += self.n_damp
```
Then the variable `ng` is not needed anymore.